app/src/androidTest/java/org/mozilla/fenix/ui/robots/TabDrawerRobot.kt

```diff
@@ -399,13 +399,12 @@ private fun threeDotMenu() = onView(withId(R.id.tab_tray_overflow))
 
 private fun assertExistingOpenTabs(title: String) {
     try {
-        mDevice.findObject(
-            UiSelector()
-                .resourceId("$packageName:id/mozac_browser_tabstray_title")
-                .textContains(title)
+        mDevice.waitNotNull(
+            Until.findObjects(By.textContains(title)), waitingTime
+        )
+        mDevice.waitNotNull(
+            Until.findObjects(By.res("$packageName:id/mozac_browser_tabstray_title")), waitingTime
         )
-            .waitForExists(waitingTime)
-
         tab(title).check(matches(isDisplayed()))
```

Comment:
The test actually failed here, at this Espresso matcher:
private fun tab(title: String) =
    onView(
        allOf(
            withId(R.id.mozac_browser_tabstray_title),
            withText(title)
        )
    )

Also sometimes fails at:
private fun assertCloseTabsButton(title: String) =
    onView(
        allOf(
            withId(R.id.mozac_browser_tabstray_close),
            withContentDescription("Close tab $title")
        )
    )
        .check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))

If you want to replace them with the findObjects selector, you can do something like:
mDevice.findObjects(By.res("$packageName:id/mozac_browser_tabstray_title")) - this returns a list of objects matching the resource name.